embeddings/pipeline/hps_pipeline.py

```diff
@@ -0,0 +1,337 @@
+        n_trials: int,
+        dataset_path: Union[str, Path, "TemporaryDirectory[str]"],
+        metric_name: str,
+        metric_key: str,
+    ):
+        self.preprocessing_pipeline = preprocessing_pipeline
+        self.evaluation_pipeline = evaluation_pipeline
+        self.pruner: optuna.pruners.BasePruner = pruner
+        self.sampler: optuna.samplers.BaseSampler = sampler
+        self.n_trials: int = n_trials
+        self.dataset_path = dataset_path
+        self.config_space = config_space
+        self.metric_name = metric_name
+        self.metric_key = metric_key
+
+    @abc.abstractmethod
+    def _get_metadata(self, parameters: SampledParameters) -> Metadata:
+        pass
+
+    @abc.abstractmethod
+    def _get_evaluation_metadata(self, parameters: SampledParameters) -> EvaluationMetadata:
+        pass
+
+    def get_best_paramaters(self, study: Study) -> Metadata:
+        best_params = study.best_params
+        parsed_params = self.config_space.parse_parameters(best_params)
+        return self._get_metadata(parsed_params)
+
+    def run(
+        self,
+    ) -> Tuple[pd.DataFrame, Metadata]:
+        self._pre_run_hook()
+        self.preprocessing_pipeline.run()
+        study: Study = optuna.create_study(
+            direction="maximize",
+            sampler=self.sampler,
+            pruner=self.pruner,
+        )
+        study.optimize(self.objective, n_trials=self.n_trials, show_progress_bar=True)
+
+        if isinstance(self.dataset_path, TemporaryDirectory):
+            self.dataset_path.cleanup()
+
+        metadata: Metadata = self.get_best_paramaters(study)
+        self._post_run_hook()
+        return study.trials_dataframe(), metadata
+
+    def objective(self, trial: optuna.trial.Trial) -> float:
+        parameters = self.config_space.sample_parameters(trial=trial)
+        parsed_params = self.config_space.parse_parameters(parameters)
+        args = self._get_evaluation_metadata(parsed_params)
+
+        pipeline = self.evaluation_pipeline(**args)
+        results = pipeline.run()
+        metric = results[self.metric_name][self.metric_key]
+        assert isinstance(metric, float)
+        return metric
+
+    def _pre_run_hook(self) -> None:
+        logging.getLogger("flair").setLevel(logging.WARNING)
+
+    def _post_run_hook(self) -> None:
+        logging.getLogger("flair").setLevel(logging.INFO)
+
+
+@dataclass
+class HuggingFaceOptimizedPipelineMetadata(ABC):
+    dataset_name: str
+    embedding_name: str
+    input_column_name: str
+    target_column_name: str
+    n_warmup_steps: int = 10
+    n_trials: int = 2
+    sample_dev_split_fraction: Optional[float] = 0.1
+    seed: int = 441
+    fine_tune_embeddings: bool = False
+    pruner_cls: Type[optuna.pruners.MedianPruner] = field(
+        init=False, default=optuna.pruners.MedianPruner
+    )
+    sampler_cls: Type[optuna.samplers.TPESampler] = field(
+        init=False, default=optuna.samplers.TPESampler
+    )
```

Comment:
should this dataclass be here or in pipelines_metadata.py file?

Reply:
It should be here, but the name may be problematic so I will rename this class.